tyndale/BibleEngine
# As a developer, I want database migrations, so I can safely deploy to production

## Summary

One of the biggest issues with deploying BibleEngine to production is the lack of database migrations. 

It's like writing code without version control. Changes to the database schema are undocumented, so we rely on TypeORM's unsafe auto-syncing, which can result in lost data in production.

Just as bad, there's no reproducible local setup for testing BibleEngine changes on a real database.

This PR accomplishes two things:

1. A reproducible local database setup
2. An initial migration which covers every schema change

This provides a framework for BibleEngine to be more reliable in production, and no more guessing 😄 

## How to Test

The sqlite migrations are tested in a CI stage I just added.

Check out the README.md updates to see how to test the local database.
